File: scala_api/src/main/scala/application/models/SummaryOfMerics/SummaryResult.scala
package application.models.SummaryOfMerics

import spray.json.{DefaultJsonProtocol, RootJsonFormat}

case class SummaryResult(
  regressionMetrics: Option[RegressionSummaryResult] = None,
  classificationMetrics: Option[ClassificationSummaryResult] = None,
  prepareDataTime: Float = 0,
  trainingModelTime: Float = 0,
  calculateMetricsTime: Float = 0
)

trait SummaryResultJsonProtocol extends DefaultJsonProtocol
  with ClassificationSummaryResultJsonProtocol
  with RegressionSummaryResultJsonProtocol {
  implicit val summaryResultFormat: RootJsonFormat[SummaryResult] = jsonFormat5(SummaryResult)
}

File: python_api/Models/SummaryResult.py
class SummaryResult:
    def __init__(self, regressionMetrics=None, classificationMetrics=None,
                 prepareDataTime=None, trainingModelTime=None, calculateMetricsTime=None,):
        self.regressionMetrics = regressionMetrics
        self.classificationMetrics = classificationMetrics
        self.prepareDataTime = prepareDataTime
        self.trainingModelTime = trainingModelTime
        self.calculateMetricsTime = calculateMetricsTime

    def setRegressionMetrics(self, regressionMetrics):
        self.regressionMetrics = regressionMetrics

    def setClassificationMetrics(self, classificationMetrics):
        self.classificationMetrics = classificationMetrics

    def setPrepareDataTime(self, time):
        self.prepareDataTime = time

    def setTrainingModelTime(self, time):
        self.trainingModelTime = time

    def setCalculateMetricsTime(self, time):
        self.calculateMetricsTime = time
